internal/app/machined/pkg/system/services/bootkube.go

```diff
@@ -327,6 +330,35 @@ func generateAssets(config runtime.Configurator) (err error) {
 		return err
 	}
 
+	// If "none" is the CNI, we expect the user to supply one or more urls that point to CNI yamls
+	if config.Cluster().Network().CNI().Name() == "none" {
+		ctx := context.Background()
+
+		for _, url := range config.Cluster().Network().CNI().URLs() {
+			urlExploded := strings.Split(url, "/")
+			fileName := urlExploded[len(urlExploded)-1]
```

Comment:
there's a nice chance it ends up empty if URL looks like `http://host/some/dir/`

idk if `path.Base()` might do better job here

Reply:
plus it might be nice to filter out `.` and `..` and other friends... or once `filepath.Clean()`, fail if `fileName` becomes empty